src/plugins/intel_cpu/src/nodes/pooling.cpp

```diff
@@ -370,28 +370,23 @@ void Pooling::prepareParams() {
                       selected_pd->getImplementationType()};
     auto engine = getEngine();
     auto builder = [&engine](const PoolingKey& key) -> executorPtr {
-        primitive_desc_iterator itpd = createDescriptorHelper(engine,
-                                                              key.inp->getDnnlDesc(),
-                                                              key.out->getDnnlDesc(),
-                                                              key.alg,
-                                                              key.stride,
-                                                              key.kernel,
-                                                              key.effective_pad_begin,
-                                                              key.effective_pad_end,
-                                                              key.effective_dilation,
-                                                              key.data_pad_end,
-                                                              key.attr);
-        dnnl::pooling_forward::primitive_desc prim_desc = itpd.get();
-        while (static_cast<bool>(itpd)) {
-            impl_desc_type impl_type = parse_impl_name(itpd.impl_info_str());
-
-            if (impl_type == key.implType) {
-                prim_desc = itpd.get();
-                break;
-            }
-            if (!itpd.next_impl())
-                break;
```

Comment:
It seems pooling primitive has a different way of handling the case when a proper implementation is not found.
It just uses the last implementation (reference), when other primitive either return nullptr or pick the first available implementation.
@dmitry-gorokhov @maxnick Does anyone know the reason. Can we pick the first available instead?

Reply:
I have changed the behavior to use the first available implementation and at least the tests which cover the case with the fallback are passed.

Reply:
Actually no. On L384 the first available implementation is stored, then if nothing matched the impl_type, the stored descriptor (witch corresponds to the first available implementation) is used.